cloudsync_onedrive.py

```diff
@@ -1200,83 +1054,58 @@ def mkdir(self, path, metadata=None) -> str:    # pylint: disable=arguments-diff
         pid = self._get_parent_id(path=path)
         log.debug("got pid %s", pid)
 
-        f = onedrivesdk.Folder()
-        i = onedrivesdk.Item()
-        _, name = self.split(path)
-        i.name = name
-        i.folder = f
-
-        with self._api() as client:
-            item = self._get_item(client, oid=pid).children.add(i)
+        _, new_folder = self.split(path)
+        api_path = self._api_path(oid=pid)
 
-        return item.id
+        log.debug("mkdir parent_path=%s", api_path)
+        data = {
+            "name": new_folder,
+            "folder": {},
+            "@microsoft.graph.conflictBehavior": "fail"
+        }
+        res = self._direct_api("post", f"{api_path}/children", json=data)
+        oid = res.get("id")
+        if not oid:
+            raise CloudFileExistsError(f"failed to mkdir: {path}")
+        return oid
 
     def delete(self, oid):
         try:
-            with self._api() as client:
-                item = self._get_item(client, oid=oid).get()
-                if not item:
-                    # I don't think this will ever happen...
-                    log.info("deleted non-existing oid %s", debug_sig(oid))  # pragma: no cover
-                    return  # file doesn't exist already...
-                info = self._info_item(item)
-                if info.otype == DIRECTORY:
-                    try:
-                        next(self.listdir(oid))
-                        raise CloudFileExistsError(f"Cannot delete non-empty folder {oid}:{info.name}")
-                    except StopIteration:
-                        pass  # Folder is empty, delete it no problem
-                self._direct_api("delete", self._get_item(client, oid=oid).api_path)
+            info = self.info_oid(oid)
+            if not info:
+                # I don't think this will ever happen...
```

Comment:
nit: IIRC, the "I don't think" comment was specific to using the SDK, so maybe doesn't apply anymore

Reply:
removed